oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/ThrottlingMetrics.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.document;
+
+/**
+ * Interface to expose throttling metrics.
+ *
+ * Concrete implementations of this interface are required to provide implementations
+ * of this class expose their throttling metrics
+ */
+public interface ThrottlingMetrics {
```

Comment:
I was wondering if this class could be split along the different concerns it seems to currently combine: config (threshold/throttlingTime) and metric (currValue).

What came to mind as an alternative would be something like
```
public interface Throttler {
  long throttlingTime();
}
```
that could replace the `ThrottlingMetrics`.

then you could move the actual algorithm that calculates the throttling time into eg a `ExponentialThrottler` that works based on an AtomicLong (for value updates) and thresholds (these 2 would move here from the current `ThrottlingMetrics`) - and similarly we could come up with different such algorithms for Throttler implementations (which could mean perhaps an interface as its input...)

and `MongoThrottlingMetrics` might either not be needed anymore or become the factory that creates an `ExponentialThrottler` with the correct inputs, taken from oplog metric etc.

This way essentially the `metrics` part is hidden from the throttling logic and become a factory/implementation detail.